### PR TITLE
feat(font-variant): Add font-variant support

### PR DESCRIPTION
**Pull request by NishanWijethunga**
This feature introduces one new property to node styles,
`fontVariant: 'superscript' | 'subscript' | 'small-caps' | 'normal'`
This style will apply to the `<Text>` components

### CODE REVIEW

File: packages/layout/src/text/processSmallCaps.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
+const processSmallCaps = fragments => {
+  const processedFragments = [];
+  fragments.forEach(({ string, attributes }) => {
+    const { fontVariant } = attributes;
+    if (fontVariant === 'small-caps') {
+      const words = string.split(' ');
+
+      const wordsList = words
```
Comment (pasinducw) [inline packages/layout/src/text/processSmallCaps.js]: Can we do this split without introducing an ASCII character into the original input?

File: packages/textkit/src/layout/index.js [hunk 2]
```diff
@@ -29,6 +30,7 @@ const layoutEngine = (engines, attributedString, container, options = {}) => {
     resolveAttachments(engines, options),
     generateGlyphs(engines, options),
     wrapWords(engines, options),
```
Comment (pasinducw) [inline packages/textkit/src/layout/index.js:32]: Can you check whether we are running into a text wrapping issue by having multiple text runs within the same word?

File: packages/layout/src/text/processSmallCaps.js (hunk 1 above)
Comment (NishanWijethunga) [reply packages/layout/src/text/processSmallCaps.js]: Yeah, fixed.

File: packages/textkit/src/layout/index.js (hunk 2 above)
Comment (NishanWijethunga) [reply packages/textkit/src/layout/index.js:32]: - I think there is no issue
image

**Pull request merged**